modules/configuration/src/main/java/org/apache/ignite/internal/configuration/ConfigurationChanger.java

```diff
@@ -274,6 +274,9 @@ public void start() throws ConfigurationChangeException {
         //Workaround for distributed configuration.
         addDefaults(superRoot);
 
+        // Validate the restored configuration.
+        validateConfiguration(superRoot, superRoot);
```

Comment:
This looks strange to me, since old values and new values will be the same. Shouldn't we use null or an empty root as the old value?